File: server/streamingplatform/src/main/java/com/mmsm/streamingplatform/video/model/VideoDetailsDto.java
package com.mmsm.streamingplatform.video.model;

import com.mmsm.streamingplatform.comment.CommentController.*;
import com.mmsm.streamingplatform.keycloak.KeycloakController.UserDto;
import com.mmsm.streamingplatform.video.videorating.VideoRatingController.*;
import lombok.Builder;
import lombok.Getter;

import java.time.Instant;
import java.util.List;

@Getter
@Builder
public class VideoDetailsDto {
    private Long id;
    private UserDto author;
    private String title;
    private String description;
    private Long upVoteCount;
    private Long downVoteCount;
    private Long viewCount;
    private Long shareCount;
    private Instant createdDate;
    private VideoRatingRepresentation currentUserVideoRating;
    private List<CommentRepresentation> directCommentDtos;
}

File: client/streamingplatform/src/app/dtos/VideoDetailsDto.ts
import {VideoRatingDto} from './VideoRatingDto';
import {CommentRepresentation} from './CommentRepresentation';
import {UserDto} from './UserDto';

export class VideoDetailsDto {
  id: number;
  author: UserDto;
  title: string;
  description: string;
  upVoteCount: number;
  downVoteCount: number;
  viewCount: number;
  shareCount: number;
  createdDate: Date;
  currentUserVideoRating: VideoRatingDto;
  directComments: CommentRepresentation[];

  constructor() {
    this.author = new UserDto();
    this.currentUserVideoRating = new VideoRatingDto();
  }
}
